patch-package
--- a/node_modules/grunt-sass-lint/tasks/sass-lint.js
+++ b/node_modules/grunt-sass-lint/tasks/sass-lint.js
@@ -7,7 +7,8 @@ module.exports = function (grunt) {
 
 	grunt.registerMultiTask('sasslint', 'Lint your Sass', function () {
 		var opts = this.options({
-				configFile: ''
+				configFile: '',
+      	warningsAreErrors: false
 			});
 		var results = [];
 
@@ -26,7 +27,7 @@ module.exports = function (grunt) {
 			} else {
 				grunt.log.writeln(resultFormat);
 			}
-      if (errorCount.count > 0) grunt.fail.warn('');
+      if ((opts.warningsAreErrors ? resultCount : errorCount.count) > 0) grunt.fail.warn('');
 		}
 	});
 };
